src/biz/bokhorst/xprivacy/XPrivacy.java

```diff
@@ -29,64 +28,107 @@
 	private static String mSecret = null;
 	private static List<String> mListHookError = new ArrayList<String>();
 	private static List<CRestriction> mListDisabled = new ArrayList<CRestriction>();
-
-	// http://developer.android.com/reference/android/Manifest.permission.html
-
-	static {
-		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP && mListDisabled.size() == 0) {
-			File disabled = new File("/data/system/xprivacy/disabled");
-			if (disabled.exists() && disabled.canRead())
-				try {
-					Log.w("XPrivacy", "Reading " + disabled.getAbsolutePath());
-					FileInputStream fis = new FileInputStream(disabled);
-					InputStreamReader ir = new InputStreamReader(fis);
-					BufferedReader br = new BufferedReader(ir);
-					String line;
-					while ((line = br.readLine()) != null)
-						if (line.length() > 0 && !line.startsWith("#")) {
-							String[] name = line.split("/");
-							if (name.length > 0) {
-								String methodName = (name.length > 1 ? name[1] : null);
-								CRestriction restriction = new CRestriction(0, name[0], methodName, null);
-								Log.w("XPrivacy", "Disabling " + restriction);
-								mListDisabled.add(restriction);
-							}
-						}
-					br.close();
-					ir.close();
-					fis.close();
-				} catch (Throwable ex) {
-					Log.w("XPrivacy", ex.toString());
-				}
-		}
-	}
-
-	// Xposed
+	private static boolean mHasErr = false;
+	
 	public void initZygote(StartupParam startupParam) throws Throwable {
 		// Check for LBE security master
 		if (Util.hasLBE()) {
 			Util.log(null, Log.ERROR, "LBE installed");
 			return;
 		}
+		
+		/*
+		 * ActivityManagerService is the beginning of the main "android" process. 
+		 * This is where the core java system is started, where the system context is created and so on. 
+		 * In pre-lollipop we can access this class directly, but in lollipop we have to visit ActivityThread first, 
+		 * since this class is now responsible for creating a class loader that can be used to access ActivityManagerService. 
+		 * It is no longer possible to do so via the normal boot class loader. Doing it like this will create a consistency between older and newer 
+		 * Android versions.
+		 * 
+		 * Note that there is no need to handle arguments in this case. And we don't need them so in case they change over time, we will simply use 
+		 * the hookAll feature. 
+		 */
+		try {
+			if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+				XposedBridge.hookAllMethods(Class.forName("android.app.ActivityThread"), "systemMain", new XC_MethodHook() {
+					@Override
+					protected void afterHookedMethod(MethodHookParam param) throws Throwable {
+						final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+						
+						try {
+							XposedBridge.hookAllConstructors(Class.forName("com.android.server.am.ActivityManagerService", false, loader), new XC_MethodHook() {
```

Comment:
In general I try to avoid using XposedBridge methods when that is not necessary, so that Xposed can easily be replaced by something else when there is a need, but since this class bridges XPrivacy and Xposed, it is not too bad here.